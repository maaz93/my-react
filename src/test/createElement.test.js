import { createElement } from '../';

describe('createElement', () => {
  test('should create simple div element', () => {
    expect(createElement('div', { id: 'test' })).toMatchSnapshot();
  });

  test('should add different attributes', () => {
    expect(
      createElement('input', { id: 'test', value: 'foo', type: 'text' })
    ).toMatchSnapshot();
  });

  test('should add text content', () => {
    expect(createElement('span', {}, 'text content')).toMatchSnapshot();
  });

  test('should create children correctly', () => {
    expect(
      createElement(
        'div',
        {},
        createElement('span', { id: 'one' }),
        createElement('a', { href: '/bar' })
      )
    ).toMatchSnapshot();
  });
});
