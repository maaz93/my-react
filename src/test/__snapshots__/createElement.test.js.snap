// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createElement should add different attributes 1`] = `
Object {
  "props": Object {
    "children": Array [],
    "id": "test",
    "type": "text",
    "value": "foo",
  },
  "type": "input",
}
`;

exports[`createElement should add text content 1`] = `
Object {
  "props": Object {
    "children": Array [
      Object {
        "props": Object {
          "children": Array [],
          "nodeValue": "text content",
        },
        "type": "TEXT_ELEMENT",
      },
    ],
  },
  "type": "span",
}
`;

exports[`createElement should create children correctly 1`] = `
Object {
  "props": Object {
    "children": Array [
      Object {
        "props": Object {
          "children": Array [],
          "id": "one",
        },
        "type": "span",
      },
      Object {
        "props": Object {
          "children": Array [],
          "href": "/bar",
        },
        "type": "a",
      },
    ],
  },
  "type": "div",
}
`;

exports[`createElement should create simple div element 1`] = `
Object {
  "props": Object {
    "children": Array [],
    "id": "test",
  },
  "type": "div",
}
`;
